hyperspy/_components/pes_voigt.py

```diff
@@ -0,0 +1,331 @@
+    transmission_function : Parameter
+    spin_orbit_splitting : Bool
+    spin_orbit_branching_ratio : float
+    spin_orbit_splitting_energy : float
+
+    Notes
+    -----
+    Uses an approximate formula according to
+    W.I.F. David, J. Appl. Cryst. (1986). 19, 63-64.
+    doi:10.1107/S0021889886089999
+    """
+
+    def __init__(self):
+        Component.__init__(self, (
+            'area',
+            'centre',
+            'FWHM',
+            'gamma',
+            'resolution',
+            'shirley_background',
+            'non_isochromaticity',
+            'transmission_function'))
+        self._position = self.centre
+        self.FWHM.value = 1
+        self.gamma.value = 0
+        self.area.value = 1
+        self.resolution.value = 0
+        self.resolution.free = False
+        self.shirley_background.free = False
+        self.non_isochromaticity.value = 0
+        self.non_isochromaticity.free = False
+        self.transmission_function.value = 1
+        self.transmission_function.free = False
+        # Options
+        self.shirley_background.active = False
+        self.spin_orbit_splitting = False
+        self.spin_orbit_branching_ratio = 0.5
+        self.spin_orbit_splitting_energy = 0.61
+        self.isbackground = False
+        self.convolved = True
+
+    def function(self, x):
+        area = self.area.value * self.transmission_function.value
+        centre = self.centre.value
+        ab = self.non_isochromaticity.value
+        if self.resolution.value == 0:
+            FWHM = self.FWHM.value
+        else:
+            FWHM = math.sqrt(self.FWHM.value ** 2 + self.resolution.value ** 2)
+        gamma = self.gamma.value
+        k = self.shirley_background.value
+        f = voigt(x,
+                  FWHM=FWHM, gamma=gamma, center=centre - ab, scale=area)
+        if self.spin_orbit_splitting is True:
+            ratio = self.spin_orbit_branching_ratio
+            shift = self.spin_orbit_splitting_energy
+            f2 = voigt(x, FWHM=FWHM, gamma=gamma,
+                       center=centre - ab - shift, scale=area * ratio)
+            f += f2
+        if self.shirley_background.active:
+            cf = np.cumsum(f)
+            cf = cf[-1] - cf
+            self.cf = cf
+            return cf * k + f
+        else:
+            return f
+
+    def estimate_parameters(self, signal, E1, E2, only_current=False):
+        """Estimate the Voigt function by calculating the momenta of the 
+        Gaussian.
+
+        Parameters
+        ----------
+        signal : Signal1D instance
+        x1 : float
+            Defines the left limit of the spectral range to use for the
+            estimation.
+        x2 : float
+            Defines the right limit of the spectral range to use for the
+            estimation.
+
+        only_current : bool
+            If False estimates the parameters for the full dataset.
+
+        Returns
+        -------
+        bool
```

Comment:
This seems like a very strange return for a method that is called estimate parameters - I understand why it is, but I think it should explain here.

Reply:
A quick search in the code shows that it is actually used: https://github.com/hyperspy/hyperspy/blob/ff2d9d38098831d0e2cde1239dd80bff056b9f67/hyperspy/signal_tools.py#L1089-L1095

As usual, documenting it would be good indeed.

Reply:
I'll add a note, if we agree on it, I can also add it to other components that are used by `remove_background`.